qiskit_experiments/analysis/curve_analysis.py

```diff
@@ -0,0 +1,690 @@
+        __fit_funcs__: List of callable to fit parameters. This is order sensitive.
+            The list index corresponds to the function index specified by __series__ definition.
+        __param_names__: Name of parameters to fit. This is order sensitive.
+        __base_fitter__: A callable to perform single curve fitting.
+            The function API should conform to the scipy curve fit module.
+
+    Examples:
+
+        T1 experiment
+        =============
+
+        In this type of experiment, the analysis deals with single curve.
+        Thus __series__ is not necessary be assigned.
+
+        .. code-block::
+
+            class AnalysisExample(CurveAnalysis):
+
+                __x_key__ = "delay"
+
+                __fit_funcs__ = [library.exponential]
+
+                __param_names__ = ["a", "tau", "b"]
+
+        IRB experiment
+        ==============
+
+        In this type of experiment, the analysis deals with two curves.
+        We need __series__ definition for each curve.
+        Both curves can be represented by the same exponential function,
+        but with different parameter set. Note that parameters will be partly shared.
+
+        .. code-block::
+
+            class AnalysisExample(CurveAnalysis):
+
+                __x_key__ = "ncliffs"
+
+                __series__ = [
+                    SeriesDef(
+                        name="standard_rb",
+                        param_names=["a", "alpha_std", "b"],
+                        fit_func_index=0,
+                        filter_kwargs={"interleaved": False}
+                    ),
+                    SeriesDef(
+                        name="interleaved_rb",
+                        param_names=["a", "alpha_int", "b"],
+                        fit_func_index=0,
+                        filter_kwargs={"interleaved": True}
+                    )
+                ]
+
+                __fit_funcs__ = [library.exponential]
+
+                __param_names__ = ["a", "alpha_std", "alpha_int", "b"]
+
+        Note that the subclass can optionally override :meth:``_post_processing``.
+        This method takes fit analysis result and calculate new entity with it.
+        EPC calculation can be performed here.
+
+        Ramsey XY experiment
+        ====================
+
+        In this type of experiment, the analysis deals with two curves.
+        We need __series__ definition for each curve.
+        In contrast to the IRB example, this experiment may have two fit functions
+        to represent cosinusoidal (real part) and sinusoidal (imaginary part) oscillation,
+        however the parameters are shared with both functions.
+
+        .. code-block::
+
+            class AnalysisExample(CurveAnalysis):
+
+                __x_key__ = "delays"
+
+                __series__ = [
+                    SeriesDef(
+                        name="x",
+                        param_names=["a", "freq", "phase", "b"],
+                        fit_func_index=0,
```

Comment:
I don't see what we gain by using this kind of reference in the first place, instead of simply passing the function itself. The `__fit_funcs` list/dict can be generated dynamically from this information, and it's one layer of refernce-complexity we are shedding.

Reply:
I think @gadial is correct. If this there is only single function, defining a key is kind of redundant setup.

Reply:
Is there a situation where we wish to serialize `__series__`? If yes, and we pass the function itself, will we still be able to serialize?

Reply:
That's good point. If your function comes from `analysis.fit_functions` module (I added this with the latest commit) we can get function name and store it in the db. We can retrieve the same function from the module later on. I'm not sure if this is proper way of serialization. Perhaps those functions should be classes to keep version information, etc...